src/classes/RD2_EntryFormController.cls

```diff
@@ -263,7 +260,36 @@ public with sharing class RD2_EntryFormController {
     @AuraEnabled
     public static String handleCommitment(String jsonRecord, String paymentMethodToken) {
         try {
-            UTIL_Http.Response response = commitmentService.handleCommitment(jsonRecord, paymentMethodToken);
+            npe03__Recurring_Donation__c rd =  (npe03__Recurring_Donation__c) JSON.deserialize(
+                jsonRecord, npe03__Recurring_Donation__c.class
+            );
+
+            commitmentService.adjustAndValidateRD(rd);
+            UTIL_Http.Response response = commitmentService.handleCommitment(rd, paymentMethodToken);
+
+            return JSON.serialize(response);
+
+        } catch (Exception ex) {
+            UTIL_AuraEnabledCommon.throwAuraHandledException(ex.getMessage());
+        }
+
+        return null;
+    }
+
+    /***
+    * @description Constructs a Recurring Donation record based on the json record, and
+    * submits the update payment commitment request to the Elevate Payments API.
+    * @param jsonRecord Recurring Donation fields as entered by the user before the record is saved
+    * @param paymentMethodToken Payment Method Token
+    * @return String Commitment API response
+    */
+    @AuraEnabled
+    public static String handleUpdatePaymentCommitment(String jsonRecord, String paymentMethodToken){
```

Comment:
New handlCommitment method to handle Payment update. The change is removing `commitmentService.adjustAndValidateRD(rd);` since when updating payment info, we do not need any pre-validation because the only info we want to send is payment token and commitment Id